### PR TITLE
Update etcd version and fix etcd register err

### PR DESCRIPTION
**Pull request by zhaoyunxing92**
**What this PR does**: 
* 更新etcd库
* 解决etcd注册中心问题（这个 #1662  pr不应该这样修改）
**Which issue(s) this PR fixes**: 
Fixes #

#1897 

**You should pay attention to items below to ensure your pr passes our ci test**
We do not merge pr with ci tests failed

- [x] All ut passed (run 'go test ./...' in project root)
- [x] After go-fmt ed , run 'go fmt project' using goland.
- [x] Golangci-lint passed, run 'sudo golangci-lint run' in project root.
- [x] After import formatted, (using [imports-formatter](https://github.com/dubbogo/tools#5-how-to-get-imports-formatter) to run 'imports-formatter .' in project root, to format your import blocks, mentioned in [CONTRIBUTING.md](https://github.com/apache/dubbo-go/blob/master/CONTRIBUTING.md) above) 
- [x] Your new-created file needs to have [apache license](https://raw.githubusercontent.com/dubbogo/resources/master/tools/license/license.txt) at the top, like other existed file does.
- [x] All integration test passed. You can run integration test locally (with docker env). Clone our [dubbo-go-samples](https://github.com/apache/dubbo-go-samples) project and replace the go.mod to your dubbo-go, and run 'sudo sh start_integration_test.sh' at root of samples project root. (M1 Slice is not Support)

### CODE REVIEW

File: CHANGE.md [hunk 1]
```diff
@@ -1,6 +1,13 @@
 # Release Notes
 ---
 
+## 1.5.8
+
+### Bugfixes
+
+- [fix bug](https://github.com/apache/dubbo-go/pull/1662)
```
Comment (AlexStocks) [inline CHANGE.md:8]: 我说过了哈，这个 title 名称你帮忙给他改下，这 fix bug 没人能看懂是啥玩意

File: remoting/etcdv3/facade.go [hunk 2]
```diff
@@ -46,54 +43,16 @@ type clientFacade interface {
...
-
 	defer r.WaitGroup().Done()
-LOOP:
 	for {
 		select {
 		case <-r.Done():
 			logger.Warnf("(ETCDV3ProviderRegistry)reconnectETCDV3 goroutine exit now...")
-			break LOOP
+			return
 			// re-register all services
```
Comment (AlexStocks) [inline remoting/etcdv3/facade.go:52]: 这块逻辑是确保一旦etcd集群发送崩溃重启后， etcd client 和 server 重连成功，不能删掉

File: remoting/etcdv3/listener.go [hunk 3]
```diff
@@ -19,38 +19,33 @@ package etcdv3
 
 import (
```
Comment (AlexStocks) [inline remoting/etcdv3/listener.go:20]: 这个 import 改的有问题吧

File: CHANGE.md [hunk 4]
```diff
@@ -27,6 +20,8 @@
 - [1.5.7 MetadataReportConfig lost the error](https://github.com/apache/dubbo-go/pull/1662)
 - [url too many colons in address](https://github.com/apache/dubbo-go/pull/1306)
 - [1.5 fix the nacos namespace from server/client.yml does not take effect](https://github.com/apache/dubbo-go/pull/1576)
```
Comment (AlexStocks) [inline CHANGE.md:22]: 把 consul 删除的 pr 也加进来

**Review by AlexStocks — APPROVED** [review]

File: CHANGE.md (hunk 4 above)
Comment (AlexStocks) [reply CHANGE.md:22]: 当然，也就是本 pr

**Review by Mulavar — APPROVED** [review]

File: registry/etcdv3/registry.go [hunk 5]
```diff
@@ -137,15 +132,14 @@ func (r *etcdV3Registry) CloseListener() {
...
-
-	return nil
+	//var tmpPath string
+	//for _, str := range strings.Split(k, "/")[1:] {
+	//	tmpPath = path.Join(tmpPath, "/", str)
+	//	if err := r.client.Put(tmpPath, ""); err != nil {
+	//		return perrors.WithMessagef(err, "create path %s in etcd", tmpPath)
+	//	}
+	//}
+	return perrors.WithMessagef(r.client.Put(k, ""), "create path %s in etcd", k)
```
Comment (justxuewei) [inline registry/etcdv3/registry.go:142]: If the above code is useless, please remove it instead of comment it.

**Review by justxuewei — APPROVED** [review]

**Review by AlexStocks — APPROVED** [review]

**Pull request merged**